src/main/java/com/pusher/client/PusherOptions.java

```diff
@@ -196,6 +198,26 @@ public String buildUrl(final String apiKey) {
                 : wsPort, apiKey, URI_SUFFIX);
     }
 
+    /**
+     *
+     * The default value is Proxy.NO_PROXY.
+     *
+     * @param proxy
+     *            Specify a proxy, e.g. <code>options.setProxy( new Proxy( Proxy.Type.HTTP, new InetSocketAddress( "proxyaddress", 80 ) ) )</code>;
+     * @return this, for chaining
+     */
+    public PusherOptions setProxy(Proxy proxy){
+        this.proxy = proxy;
```

Comment:
You're right: https://github.com/TooTallNate/Java-WebSocket/blob/d964307fa0c7deea6f93a681f5b1db3812f30370/src/main/java/org/java_websocket/client/WebSocketClient.java#L365

We could just throw the same exception here.